src/app/shared/utils/utils.ts

```diff
@@ -326,6 +326,17 @@ export function deepDiffObjects<T extends Object, U extends Object>(a: T, b: U)
   return diff(a, b) as RecursivePartial<T | U>;
 }
 
+/** @returns the keys from the first object that are not present in the second object. */
+export function shallowDiffObjectKeys<T extends Object, U extends Object>(a: T, b: U) {
```

Comment:
nit(blocking) - I don't think the name is very indicative that it only returns list of keys that appear in A but not in B (i.e. removed from A) - would be better to show both added and removed. I'll add a quick commit to update

Reply:
Updated with 926f438